R/compute_centiloid.R

```diff
@@ -0,0 +1,176 @@
+#' Compute Centiloid Value
+#'
+#' Computes centiloid values based on amyloid PET tracer, SUVR value and pipeline,
+#' and reference region.
+#' Also allows for custom formula parameters.
+#'
+#' @param tracer Amyloid PET tracer from ag_neuro dataset
+#'
+#'   A character string is expected.
+#'   When custom parameters are not provided, accepted values are:
+#'   '18F-Florbetapir', '18F-Florbetaben'
+#'   If custom slope and intercept are provided, any string is accepted.
+#'
+#' @param pipeline SUVR pipeline from nv_neuro dataset
+#'
+#'   A character string is expected.
+#'   When custom parameters are not provided, accepted values are:
+#'   'AVID FBP SUVR PIPELINE', 'BERKELEY FBP SUVR PIPELINE', 'BERKELEY FBB SUVR PIPELINE'
+#'   If custom slope and intercept are provided, any string is accepted.
+#'
+#' @param ref_region Reference region from suppnv_neuro dataset
+#'
+#'   A character string is expected.
+#'   When custom parameters are not provided, accepted values are:
+#'   'Whole Cerebellum', 'Composite Reference Region'
+#'   If custom slope and intercept are provided, any string is accepted.
+#'
+#' @param suvr SUVR value
+#'
+#'   A numeric value is expected.
+#'
+#' @param custom_slope Optional slope parameter for custom centiloid calculation formula
+#'
+#'   A numeric value is expected when provided.
+#'   When provided with custom_slope, this overrides the standard formula parameters.
+#'   Default is NULL.
+#'
+#' @param custom_intercept Optional intercept parameter for custom centiloid calculation formula
+#'
+#'   A numeric value is expected when provided.
+#'   When provided with custom_intercept, this overrides the standard formula parameters.
+#'   Default is NULL.
+#'
+#' @details
+#' The centiloid scale is a standardized quantitative measure for amyloid PET imaging
+#' that allows comparison between different tracers and analysis methods. This function
+#' converts SUVR values to the centiloid scale based on published conversion
+#' equations for specific tracer, pipeline, and reference region combinations.
+#'
+#' Centiloid is calculated as:
+#'
+#' Centiloid = slope * SUVR + intercept
```

Comment:
This would look nicer if formatted like an equation (see [here](https://github.com/pharmaverse/admiral/blob/976de6a852ca94e68f190ea64cd36d4b998db435/R/compute_kidney.R#L52) for an example)

```suggestion
#' \deqn{Centiloid = slope \times SUVR + intercept}
```